pkg/deploy/lattice/targets_manager.go

```diff
@@ -50,6 +50,12 @@ func (s *defaultTargetsManager) Update(ctx context.Context, modelTargets *model.
 			modelTg.ID(), modelTargets.Spec.StackTargetGroupId)
 	}
 
+	// Only take care of pods that are ready, for backwards compatibility.
+	// TODO: Pod readiness support.
+	modelTargets.Spec.TargetList = utils.SliceFilter(modelTargets.Spec.TargetList, func(t model.Target) bool {
+		return t.Ready
+	})
```

Comment:
We should have a unit test that ensures we filter out items that aren't ready. Happy to have this as a follow up.

Reply:
Though I think the readiness support means we won't do this filter anymore anyway.